src/analytics/ledger/address_balance.rs

```diff
@@ -3,28 +3,24 @@
 
 use std::collections::HashSet;
 
-use super::TransactionAnalytics;
+use super::{AddressCount, TransactionAnalytics};
 use crate::types::{
-    ledger::{LedgerOutput, LedgerSpent},
+    ledger::{LedgerOutput, LedgerSpent, MilestoneIndexTimestamp},
     stardust::block::Address,
-    tangle::MilestoneIndex,
 };
 
-/// The number of addresses.
-pub struct AddressCount(usize);
-
 /// Computes the number of addresses the currently hold a balance.
 pub struct AddressBalanceAnalytics {
     addresses: HashSet<Address>,
```

Comment:
This is definitely not right. There's no way to know if they hold a balance without holding said balance in the map and subtracting out transaction inputs.